test/test_compare.py

```diff
@@ -2,647 +2,258 @@
-        # ),
-        # (
-        #     "Graphics[{Disk[{0,0},1]}] == Compile[{x}, Sqrt[x]]",
-        #     "Graphics[{Disk[{0,0},1]}] == Compile[{x}, Sqrt[x]]",
-        # ),
-        # ("Compile[{x}, Sqrt[x]] == Compile[{x}, Sqrt[x]]", "True"),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == Graphics[{Disk[{0,0},1]}]",
-        #     "Compile[{x}, Sqrt[x]] == Graphics[{Disk[{0,0},1]}]",
-        # ),
-        ("1 / 4 == Sqrt[I] Infinity", "False"),
-        ("1 / 4 == 2.+ I Pi", "False"),
-        ("1 / 4 == 3+I Pi", "False"),
-        (
-            '1 / 4 == TestFunction["Tengo una vaca lechera"]',
-            '1 / 4 == TestFunction["Tengo una vaca lechera"]',
-        ),
-        ("-Infinity == Sqrt[I] Infinity", "False"),
-        (
-            'TestFunction["Tengo una vaca lechera"] == "1 / 4"',
-            'TestFunction["Tengo una vaca lechera"] == 1 / 4',
-        ),
-        (
-            'TestFunction["Tengo una vaca lechera"] == 0',
-            'TestFunction["Tengo una vaca lechera"] == 0',
-        ),
-        (
-            'TestFunction["Tengo una vaca lechera"] == 1 / 4',
-            'TestFunction["Tengo una vaca lechera"] == 1 / 4',
-        ),
-        # (
-        #     'TestFunction["Tengo una vaca lechera"] == Compile[{x}, Sqrt[x]]',
-        #     'TestFunction["Tengo una vaca lechera"] == Compile[{x}, Sqrt[x]]',
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == 2 + 3 a",
-        #     "Compile[{x}, Sqrt[x]] == 2 + 3 a",
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == Infinity",
-        #     "Compile[{x}, Sqrt[x]] == Infinity",
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == -Infinity",
-        #     "Compile[{x}, Sqrt[x]] == -Infinity",
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == Sqrt[I] Infinity",
-        #     "Compile[{x}, Sqrt[x]] == Sqrt[I] Infinity",
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == a",
-        #     "Compile[{x}, Sqrt[x]] == a",
-        # ),
-        # (
-        #     'Compile[{x}, Sqrt[x]] == "a"',
-        #     'Compile[{x}, Sqrt[x]] == "a"',
-        # ),
-        # (
-        #     'Compile[{x}, Sqrt[x]] == "1 / 4"',
-        #     'Compile[{x}, Sqrt[x]] == "1 / 4"',
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == 0",
-        #     "Compile[{x}, Sqrt[x]] == 0",
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == 0",
-        #     "Compile[{x}, Sqrt[x]] == 0",
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == 0.25",
-        #     "Compile[{x}, Sqrt[x]] == 0.25",
-        # ),
-        # (
-        #     "Compile[{x}, Sqrt[x]] == Sqrt[2]",
-        #     "Compile[{x}, Sqrt[x]] == Sqrt[2]",
-        # ),
-    ],
+    ("str_lhs", "str_rhs", "str_expected"),
+    tests1,
 )
-def test_cmp1_no_pass(str_expr, str_expected):
-    check_evaluation(str_expr, str_expected)
+def test_cmp1_no_pass(str_lhs, str_rhs, str_expected):
```

Comment:
Overall I really like how this test looks. Thanks!

One small thing is that I would add a comment here that we assume that == is commutative and test that this is so. 

I similar comment probably should go in the user-visible doc too.